sql/catalog/lease: TestLeaseWithOfflineTables failed

sql/catalog/lease.TestLeaseWithOfflineTables [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18551688?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18551688?buildTab=artifacts#/) on master @ [1aab60e22070cad7931b5def40ec68b0700141b5](https://github.com/cockroachdb/cockroach/commits/1aab60e22070cad7931b5def40ec68b0700141b5):


```
=== RUN   TestLeaseWithOfflineTables
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/5b16c701c1da9e1ccd7b3d919dbf90f6/logTestLeaseWithOfflineTables613656683
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:2601: 
        	Error Trace:	pkg/sql/catalog/lease/lease_test.go:2601
        	            				pkg/sql/catalog/lease/lease_test.go:2624
        	Error:      	Should be false
        	Test:       	TestLeaseWithOfflineTables
        	Messages:   	lease should have been present but wasn't
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b16c701c1da9e1ccd7b3d919dbf90f6/logTestLeaseWithOfflineTables613656683
--- FAIL: TestLeaseWithOfflineTables (1.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseWithOfflineTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46676

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -2574,7 +2574,13 @@ func TestLeaseWithOfflineTables(t *testing.T) {
 			desc.State = next
 			return descsCol.WriteDesc(ctx, false /* kvTrace */, desc, txn.KV())
 		}))
-
+		// Wait for the initial version of the descriptor if it's going from DROPPED
+		// to public. This transition doesn't occur in the real world, so the the
+		// descriptor txn will not wait for the initial version.
+		if expected == descpb.DescriptorState_DROP && next == descpb.DescriptorState_PUBLIC {
+			require.NoError(t,
+				execCfg.LeaseManager.WaitForInitialVersion(ctx, testTableID(), retry.Options{}, nil))
+		}
 		// Wait for the lease manager's refresh worker to have processed the
 		// descriptor update.
 		<-blockDescRefreshed

